hack/make-rules/run-e2e-tests.sh

```diff
@@ -25,20 +25,119 @@ source "${YURT_ROOT}/hack/lib/build.sh"
 readonly LOCAL_ARCH=$(go env GOHOSTARCH)
 readonly LOCAL_OS=$(go env GOHOSTOS)
 readonly YURT_E2E_TARGETS="test/e2e/yurt-e2e-test"
-KUBECONFIG=${KUBECONFIG:-${HOME}/.kube/config}
+readonly EDGE_AUTONOMY_NODES_NUM=3
+cloudNodeContainerName="openyurt-e2e-test-control-plane"
+edgeNodeContainerName="openyurt-e2e-test-worker"
+edgeNodeContainer2Name="openyurt-e2e-test-worker2"
+KUBECONFIG=$HOME/.kube/config
+
+
+function set_flags() {
+    goldflags="${GOLDFLAGS:--s -w $(project_info)}"
+    gcflags="${GOGCFLAGS:-}"
+    goflags=${GOFLAGS:-}
+
+    target_bin_dir=$(get_binary_dir_with_arch ${YURT_LOCAL_BIN_DIR})
+    mkdir -p ${target_bin_dir}
+    cd ${target_bin_dir}
+    echo "Building ${YURT_E2E_TARGETS}"
+    testpkg="$(dirname ${YURT_E2E_TARGETS})"
+    filename="$(basename ${YURT_E2E_TARGETS})"
+#    set kubeconfig
+    docker exec -d $edgeNodeContainerName /bin/bash -c 'mkdir /root/.kube/ -p'
+    docker exec -d $edgeNodeContainerName /bin/bash -c  "echo 'export KUBECONFIG=/root/.kube/config' >> /etc/profile && source /etc/profile" 
+    docker cp $KUBECONFIG $edgeNodeContainerName:/root/.kube/config
+}
+
+# set up flannel
+function setUpFlannel() {
+    local flannelYaml="https://raw.githubusercontent.com/flannel-io/flannel/master/Documentation/kube-flannel.yml"
+    local flannelDs="kube-flannel-ds"
+    local flannelNameSpace="kube-flannel"
+    local POD_CREATE_TIMEOUT=120s
+    curl -o ${YURT_ROOT}/flannel.yaml $flannelYaml
+    kubectl apply -f ${YURT_ROOT}/flannel.yaml
+    # check if flannel on every node is ready, if so, "daemon set "kube-flannel-ds" successfully rolled out"
+    kubectl rollout status daemonset kube-flannel-ds -n kube-flannel --timeout=${POD_CREATE_TIMEOUT}
+
+    # set up bridge cni plugins for every node
+    wget -O ${YURT_ROOT}/cni.tgz https://github.com/containernetworking/plugins/releases/download/v1.1.1/cni-plugins-linux-amd64-v1.1.1.tgz
+
+    docker cp ${YURT_ROOT}/cni.tgz $cloudNodeContainerName:/opt/cni/bin/
+    docker exec -t $cloudNodeContainerName /bin/bash -c 'cd /opt/cni/bin && tar -zxf cni.tgz'
+
+    docker cp ${YURT_ROOT}/cni.tgz $edgeNodeContainerName:/opt/cni/bin/
+    docker exec -t $edgeNodeContainerName /bin/bash -c 'cd /opt/cni/bin && tar -zxf cni.tgz'
+
+    docker cp ${YURT_ROOT}/cni.tgz $edgeNodeContainer2Name:/opt/cni/bin/
+    docker exec -t $edgeNodeContainer2Name /bin/bash -c 'cd /opt/cni/bin && tar -zxf cni.tgz'    
+}
+
+# install gingko
+function getGinkgo() {
+    go install github.com/onsi/ginkgo/v2/ginkgo@v2.3.0
+    go get github.com/onsi/gomega/...
+    go mod tidy
```

Comment:
We only need the ginkgo binary, so do not `go get gomega` and `go mod tidy`.
Also, the ginkgo version we use is v2.1.4

Reply:
will do